### ISSUE TITLE
[v4] | [v2] Internal `BottomSheetScrollView` disappears when the state of the component changes if `enableDynamicSizing` is true

### ISSUE THREAD

**Issue by khzouroussama** [gorhom/react-native-scroll-bottom-sheet#1524]
# Bug

The `BottomSheetScrollView` (not a direct child of the BottomSheetModal) disappears whenever the state of the component changes in the latest version with the new `enableDynamicSizing` prop enabled. Please see the attached video

https://github.com/gorhom/react-native-bottom-sheet/assets/20207933/50454e66-a004-4313-9d4b-36f3727aebff

## Environment info

| Library                         | Version |
| ------------------------------- | ------- |
| @gorhom/bottom-sheet            | 4.5.0   |
| react-native                    | 0.68.1   |
| react-native-reanimated         | 2.9.1   |
| react-native-gesture-handler    | 2.5.0   |

## Steps To Reproduce

1. create a bottomSheetModal with `enableDynamicSizing` according to the example in the `/example/app/screens/modal/DynamicSnapPointExample.tsx`
2. Wrap the children of the  a BottomSheetScrollView component inside 
3. Make sure the BottomSheetScrollView component is within BottomSheetView

Describe what you expected to happen:

1. The content of the scrollView to be show at all times
2. heigh to be calculated correctly

## Reproducible sample code
(modified the example code)

```tsx
import React, { useCallback, useMemo, useRef, useState } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import {
  BottomSheetModal,
  BottomSheetScrollView,
  BottomSheetView,
} from '@gorhom/bottom-sheet';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { Button } from '../../components/button';
import { withModalProvider } from './withModalProvider';

const DynamicSnapPointExample = () => {
  // state
  const [count, setCount] = useState(0);

  // hooks
  const { bottom: safeBottomArea } = useSafeAreaInsets();
  const bottomSheetRef = useRef<BottomSheetModal>(null);

  // callbacks
  const handleIncreaseContentPress = useCallback(() => {
    setCount(state => state + 1);
  }, []);
  const handleDecreaseContentPress = useCallback(() => {
    setCount(state => Math.max(state - 1, 0));
  }, []);

  const handlePresentPress = useCallback(() => {
    bottomSheetRef.current?.present();
  }, []);
  const handleDismissPress = useCallback(() => {
    bottomSheetRef.current?.dismiss();
  }, []);

  // styles
  const contentContainerStyle = useMemo(
    () => ({
      ...styles.contentContainerStyle,
      paddingBottom: safeBottomArea || 6,
    }),
    [safeBottomArea]
  );
  const emojiContainerStyle = useMemo(
    () => ({
      ...styles.emojiContainer,
      height: 50 * count,
    }),
    [count]
  );

  // renders
  return (
    <View style={styles.container}>
      <Button label="Present" onPress={handlePresentPress} />
      <Button label="Dismiss" onPress={handleDismissPress} />
      <BottomSheetModal
        ref={bottomSheetRef}
        enableDynamicSizing={true}
        enablePanDownToClose={true}
      >
        <BottomSheetView style={contentContainerStyle}>
          <BottomSheetScrollView>                          {/**  <---------- add a the scrollView here **/}
            <Text style={styles.message}>
              Could this sheet modal resize to its content height ?
            </Text>
            <View style={emojiContainerStyle}>
              <Text style={styles.emoji}>😍</Text>
            </View>
            <Button label="Yes" onPress={handleIncreaseContentPress} />
            <Button label="Maybe" onPress={handleDecreaseContentPress} />
          </BottomSheetScrollView>
        </BottomSheetView>
      </BottomSheetModal>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 24,
  },
  contentContainerStyle: {
    paddingTop: 12,
    paddingHorizontal: 24,
    backgroundColor: 'white',
  },
  message: {
    fontSize: 24,
    fontWeight: '600',
    marginBottom: 12,
    color: 'black',
  },
  emoji: {
    fontSize: 156,
    textAlign: 'center',
    alignSelf: 'center',
  },
  emojiContainer: {
    overflow: 'hidden',
    justifyContent: 'center',
  },
});

export default withModalProvider(DynamicSnapPointExample);
```

**Comment by antoine-waag** [comment]
@khzouroussama Since I set the props `enableDynamicSizing` to `true`, I've got the same problem with the `BottomSheetView` (I don't use `BottomSheetScrollView`).

If I update a state in a child component, some content disappears..

**Comment by gorhom** [comment]
i see that you wrapped a scrollable with a view, this won't enable the content dynamic sizing, so you will need to remove the `BottomSheetView` 

```
        <BottomSheetView style={contentContainerStyle}> <---- remove this layer
          <BottomSheetScrollView>     
```

**Comment by khzouroussama** [comment]
Yes @gorhom that's correct, it would work if the `BottomSheetScrollView` is the direct child,
I am not sure if this is an excepted behavior, but the react native scrollView worked fine (except for missing integration with the modal pan down to close gesture 😞)

**Comment by gorhom** [comment]
good catch,,, i fixed it just now with v4.5.1

**Comment by khzouroussama** [comment]
Thanks @gorhom, amazing work as always 🙏
